### PR TITLE
feat: trace handler (CORE-5344)

### PR DESCRIPTION
**Pull request by leartgjoni-voiceflow**
to-test:
- import [test trace block.zip](https://github.com/voiceflow/general-runtime/files/6111400/test.trace.block.zip)

manual tests performed:
- default path (doesn't stop)
- inline events (event1, event2)
- global jump events (event3, event4)
- global push event (help)
- hit intent (travel_intent) from trace block
- hit command (second_flow) from trace block

### CODE REVIEW

**Review by DecathectZero — APPROVED** [review]

File: lib/services/runtime/handlers/event/index.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+import { GeneralRuntime, isGeneralRequest, isIntentRequest } from '@/lib/services/runtime/types';
 
 import { mapEntities } from '../../utils';
 
 export const intentEventMatcher = {
-  match: (context: { runtime: GeneralRuntime; event: GeneralEvent | null }): context is { runtime: Runtime<IntentRequest>; event: IntentEvent } => {
+  match: (context: { runtime: GeneralRuntime; event: Event | null }): context is { runtime: Runtime<IntentRequest>; event: IntentEvent } => {
     const request = context.runtime.getRequest();
     if (!isIntentRequest(request)) return false;
     if (context.event?.type !== EventType.INTENT) return false;
```
Comment (DecathectZero) [inline lib/services/runtime/handlers/event/index.ts:13]: I guess we can do an intent event typeguard here too, instead of casting `as`

Comment (DecathectZero) [reply lib/services/runtime/handlers/event/index.ts:13]: nevermind, I guess the whole point of this function is to be a typeguard

File: lib/services/runtime/handlers/trace.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import CommandHandler from './command';
+import { findEventMatcher } from './event';
+
+const utilsObj = {
+  commandHandler: CommandHandler(),
+  findEventMatcher,
+};
+
+export const TraceHandler: HandlerFactory<Node, typeof utilsObj> = (utils) => ({
+  canHandle: (node) => !!node._v,
```
Comment (DecathectZero) [inline lib/services/runtime/handlers/trace.ts:15]: I would do `_v === 1` so its even more explicit

File: lib/services/nlu/index.ts [hunk 3]
```diff
@@ -69,7 +69,7 @@ class NLU extends AbstractManager<{ utils: typeof utils }> implements ContextHan
     }
 
     const request = await this.predict({
-      query: context.request.payload,
+      query: (context.request.payload as unknown) as string,
```
[the remark this answers is not in the document]
Comment (leartgjoni-voiceflow) [reply lib/services/nlu/index.ts:72]: Yea I've tried to create a typeguard but the types are not compatible.

Screen Shot 2021-03-10 at 9 35 40 AM

cause payload is `Record<string, unknown>`, I guess we could change it to just `unknown`? @DecathectZero

[the remark this answers is not in the document]
Comment (DecathectZero) [reply lib/services/nlu/index.ts:72]: yeah, since TextRequest is actually just `payload: string`

**Pull request merged**